driver/git.go

```diff
@@ -161,6 +167,20 @@ func (driver *GitDriver) setUpRepo() error {
 	return nil
 }
 
+func (driver *GitDriver) setUpAuth() error {
+	err := driver.setUpKey()
```

Comment:
I feel like this should be gated by when driver.PrivateKey is set.

Similarly, the check that's done inside of setUpUsernamePassword on line 205 should be moved up here.

Something like:

```go
if driver.PrivateKey != "" {
  err := driver.setUpKey()
  if err != nil {
    return err
  }
}

if driver.Username != "" && driver.Password != "" {
  err = driver.setUpUsernamePassword()
  if err != nil {
    return err
  }
}
```

This makes it so we won't write a private key that is empty, and makes setUpUsernamePassword less surprising.